File: medicines-frontend/src/app/models/MedicationResult.ts
import {MedicationDto} from "./MedicationDto";
import {DayPeriod} from "./DayPeriod";
import {TemporaryMedicationDto} from "./TemporaryMedicationDto";

export interface MedicationResult {
    dayPeriod: DayPeriod;
    medications: MedicationDto[];
    temporaryMedications: TemporaryMedicationDto[];
}

File: medicines/src/main/java/com/nr/medicines/models/dto/MedicationResult.java
package com.nr.medicines.models.dto;

import com.nr.medicines.models.DayPeriod;
import lombok.Builder;
import lombok.Data;

import java.util.List;

@Data
@Builder
public class MedicationResult {
    private DayPeriod dayPeriod;
    private List<MedicationDto> medications;
    private List<TemporaryMedicationDto> temporaryMedications;
}
